src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -2529,23 +2529,81 @@ public static ArgumentSyntax Argument(ExpressionSyntax expression)
         /// <summary>Creates a new ArgumentSyntax instance.</summary>
         public static ArgumentSyntax Argument(NameColonSyntax nameColon, SyntaxToken outKeyword, VariableDeclarationSyntax declaration)
         {
-            if (outKeyword.Kind() != SyntaxKind.OutKeyword)
+            return Argument(nameColon, outKeyword, (CSharpSyntaxNode)declaration);
```

Comment:
Same thing for the overload that takes a single syntax node, which I think refers to the one on line 2524 a few lines above.
> Factory static ArgumentSyntax Argument(CSharpSyntaxNode expressionOrDeclaration) will be internal.